.github/workflows/pr-check.yml

```
@@ -107,6 +108,16 @@ jobs:
         with:
           python-version: ${{ matrix.python }}
 
+      - name: Install specific pytest version
+        if: matrix.pytest-version == 'pytest@pre-release'
+        run: |
+          python -m pip install --pre pytest
+
+      - name: Install specific pytest version
+        if: matrix.pytest-version != 'pytest@pre-release'
+        run: |
+          python -m pip install "${{ matrix.pytest-version }}"
+
```

```suggestion

      - name: Install specific pytest version
        run: python -m pytest --version
```

probably a good idea to print version